sky/cli.py

```diff
@@ -2675,25 +2676,26 @@ def _hint_or_raise_for_down_jobs_controller(controller_name: str):
     controller = controller_utils.Controllers.from_name(controller_name)
     assert controller is not None, controller_name
 
+    try:
+        with rich_utils.safe_status(
+                ux_utils.spinner_message('Checking controller status')):
+            handle = backend_utils.is_controller_accessible(controller,
+                                                            stopped_message='')
+    except exceptions.ClusterNotUpError as e:
+        if e.cluster_status is None:
+            click.echo('Controller has already been torn down.')
+            sys.exit(0)
+
+        spinner_msg = (f'Controller is {e.cluster_status.value} - restarting '
+                       'to check that it\'s safe to terminate.')
+        with rich_utils.safe_status(ux_utils.spinner_message(spinner_msg)):
+            with sky_logging.silent():
+                handle = sky.start(controller_name)
```

Comment:
Why do we check the controller status? If the main purpose is to start the controller in INIT state, we have experienced some issue when the controller stucked in an INIT state due to some issue with the code, and trying to restart the controller may not succeed.

minor: we should keep a single spinner context and use a `status.update` for updating the status. `logger.info` will automatically handle printing message when spinner is on.

Reply:
It's possible the controller is STOPPED but still has clusters in the db. We need to catch this case.

I can refactor the spinner. I guess we should avoid click.echo in that case?

Reply:
In that case, should we just avoid autostop if there are clusters in the cluster table that are not terminated?